apps/backend/app/services/job_service.py

```
@@ -12,125 +14,364 @@
+    - Concurrent processing with memory limits
+    - Streaming job parsing
+    """
+    
     def __init__(self, db: AsyncSession):
         self.db = db
         self.json_agent_manager = AgentManager(model="gemma3:4b")
+        self.keyword_extractor = EfficientKeywordExtractor()
+        self.tokenizer = MemoryEfficientTokenizer()
+        self._extraction_cache = {}
+        
+    @lru_cache(maxsize=100)
```

Ruff (0.11.9)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)